truffle/contracts/Splitter.sol

```diff
@@ -13,20 +15,22 @@ contract Splitter is Stoppable {
     
     function splitEth(address receiver1, address receiver2) public _onlyIfRunning payable returns(bool) {
         uint half = msg.value / 2;
-        
+
         if (msg.value % 2 > 0) {
-            balances[msg.sender]++;
+            balances[msg.sender] = balances[msg.sender].add(1);
         }
         
-        balances[receiver1] += half;
-        balances[receiver2] += half;
+        balances[receiver1] = balances[receiver1].add(half);
+        balances[receiver2] = balances[receiver2].add(half);
 
         emit LogEthSent(msg.sender, msg.value, receiver1, receiver2);
         
         return true;
     }
     
     function withdraw() public returns(bool) {
+        require(balances[msg.sender] > 0);
+
         uint amount = balances[msg.sender];
```

Comment:
This is the second time you `SLOAD` `balances[msg.sender]`. It has not changed. Can you save some gas?

Reply:
I can indeed, place the ```uint amount = ...``` above the **require** but, it will cost more gas if the user does not have a balance > 0

Reply:
True, the rare case will cost more.

Reply:
So what will be the best approach?

Reply:
To move `uint amount =` above like you said. You care less about those who use the contract wrongly.